src/analyses/invariant_propagation.cpp

```diff
@@ -100,8 +100,8 @@ void invariant_propagationt::get_objects_rec(
 
     const struct_typet::componentst &c=struct_type.components();
 
-    exprt member_expr(ID_member);
-    member_expr.copy_to_operands(src);
+    member_exprt member_expr;
```

Comment:
No need to do it as part of this PR, but seems like would make more sense to introduce this _inside_ the for loop then you can construct from the `component` directly.

Reply:
But you should probably use `set_component_name`  on line 108 here